### PR TITLE
Nowcast selective merge

### PR DESCRIPTION
**Pull request by cgsandford**
Merges nowcasting feature branch changes into master.

**This PR does not require line-by-line review**.   Everything before the last merge from master went through 2 formal reviews before going into the nowcasting feature branch.  The last few commits (which do need reviewing) cover:

- Removal of unneeded functionality (pysteps interfacing and the "optical flow perturbation" function and CLI)
- Rearrangement of acceptance test data to replace the existing "master" structure, requiring changes to paths in 5 acceptance test files

The reviewer should check that the correct features have been retained, and that the acceptance tests run on the new data (see associated JIRA ticket).

### CODE REVIEW

File: improver_tests/nowcasting/optical_flow/__init__.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Set up function for dummy cube with correct metadata"""
+
+import numpy as np
+
+from improver.synthetic_data.set_up_test_cubes import set_up_variable_cube
+
+
+def set_up_test_cube(data, name, units, time, gridsize=30):
```
Comment (benfitzpatrick) [inline improver_tests/nowcasting/optical_flow/__init__.py:38]: Codacy found an issue: [Unused argument 'gridsize'](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6224167)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- improver/cli/nowcast_optical_flow_from_winds.py  2
- improver_tests/nowcasting/optical_flow/test_utilities.py  3
- improver_tests/acceptance/test_nowcast_optical_flow_from_winds.py  3
- improver_tests/nowcasting/pysteps_advection/test_PystepsExtrapolate.py  4
- improver/nowcasting/pysteps_advection.py  5
- improver/utilities/redirect_stdout.py  1
- improver_tests/nowcasting/optical_flow/test_generate_advection_velocities_from_winds.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6224167&bid=20286440)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/nowcasting/optical_flow/test_generate_advection_velocities_from_winds.py  2
- improver/nowcasting/pysteps_advection.py  5
- improver/utilities/redirect_stdout.py  1
- improver_tests/acceptance/test_nowcast_optical_flow_from_winds.py  3
- improver_tests/nowcasting/optical_flow/test_utilities.py  3
- improver/cli/nowcast_optical_flow_from_winds.py  2
- improver_tests/nowcasting/pysteps_advection/test_PystepsExtrapolate.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6224167&bid=20286440)

**Review by MoseleyS — APPROVED** [review]
I'm happy that these changes are as intended and that the tests pass.

**Review by bayliffe — APPROVED** [review]
This looks good to me. The tests run without any requirements for additional paths (beyond the usual) due to the relevant parts of pysteps being available in the ppdev local build space.

**Pull request merged**